docs/User-guide.rst

```diff
@@ -0,0 +1,73 @@
+**********
+User Guide
+**********
+
+Once you understand :ref:`how dedupe works <how-it-works-label>`, and you've taken
+a look at some of the :doc:`Examples`,
+you probably want to figure out how to adapt dedupe to your own problem. This
+user guide is for that.
+
+Memory Considerations
+=====================
+
+The most common memory bottleneck is during clustering. Within every block,
+after computing a score for each pair of records, dedupe then runs a
+connected-components algorithm as a pre-processing step for the hierarchical
+agglomerative clustering.
+`Currently, this requires having the entire edge list in memory
+<https://github.com/dedupeio/dedupe/issues/819>`__, which for :math:`N`` records
+leads to :math:`O(N^2)` memory usage. If you end up with any "superblocks" because
+your blocking rules weren't picky enough, this can run you out of memory. You
```

Comment:
super blocks won't really create a memory problem. they could make things too slow, but we filter the components to not be bigger than 30,000 vertices

Reply:
OK I can just remove this